packages/ui-app/src/Params/Param/Balance.tsx

```diff
@@ -0,0 +1,192 @@
+// Copyright 2017-2018 @polkadot/ui-app authors & contributors
+// This software may be modified and distributed under the terms
+// of the ISC license. See the LICENSE file for details.
+
+import { ApiProps } from '@polkadot/ui-react-rx/types';
+import { I18nProps } from '../../types';
+import { Props as BareProps } from '../types';
+
+import BN from 'bn.js';
+import React from 'react';
+import withApi from '@polkadot/ui-react-rx/with/api';
+
+import isValidBalance from '../../util/isValidBalance';
+import Bare from './Bare';
+import InputNumber from '../../InputNumber';
+import translate from '../../translate';
+
+type Props = I18nProps & ApiProps & BareProps;
+
+type State = {
+  error?: React.ReactNode,
+  info?: React.ReactNode
+};
+
+class Balance extends React.PureComponent<Props, State> {
+  state: State;
+
+  constructor (props: Props) {
+    super(props);
+
+    this.state = {
+      error: '',
+      info: ''
+    };
+  }
+
+  render () {
+    const { className, defaultValue: { value }, isError, label, style, t, withLabel } = this.props;
+    const { error, info } = this.state;
+    const defaultValue = new BN((value as BN).toString(10) || '0').toString(10);
+
+    const maxLengthForLatestChainSpec = 39;
```

Comment:
This is going to create issues. :( For InputNumber, pass through the bitLength and calculate the number there. i.e. 

```
const maxValue = new BN(2).pow(bitLength).subn(1);
const maxLength = maxValue.toString().length;`
```

Additionally, the input value can now actually be checked against this maxValue to see that if it is still indeed in-range.